csharp/src/Apache.Arrow/Arrays/Time64Array.cs

```diff
@@ -0,0 +1,172 @@
+            : this(new ArrayData(type, length, nullCount, offset,
+                new[] { nullBitmapBuffer, valueBuffer }))
+        { }
+
+        public Time64Array(ArrayData data)
+            : base(data)
+        {
+            data.EnsureDataType(ArrowTypeId.Time64);
+        }
+
+        public override void Accept(IArrowArrayVisitor visitor) => Accept(this, visitor);
+
+
+        /// <summary>
+        /// Get the time at the specified index as seconds
+        /// </summary>
+        /// <param name="index">Index at which to get the date.</param>
+        /// <returns>Returns a long, or <c>null</c> if there is no object at that index.
+        /// </returns>
+        public long? GetSeconds(int index)
+        {
+            long? value = GetValue(index);
+            if (value == null)
+            {
+                return null;
+            }
+
+            var unit = ((Time64Type) Data.DataType).Unit;
+            return unit switch
+            {
+                TimeUnit.Second => value,
+                TimeUnit.Millisecond => value / 1_000,
+                TimeUnit.Microsecond => value / 1_000_000,
+                TimeUnit.Nanosecond => value / 1_000_000_000,
+                _ => throw new InvalidDataException($"Unsupported time unit for Time64Type: {unit}")
+            };
+        }
+
+        /// <summary>
+        /// Get the time at the specified index as milliseconds
+        /// </summary>
+        /// <param name="index">Index at which to get the date.</param>
+        /// <returns>Returns a long, or <c>null</c> if there is no object at that index.
+        /// </returns>
+        public long? GetMilliSeconds(int index)
+        {
+            long? value = GetValue(index);
+            if (value == null)
+            {
+                return null;
+            }
+
+            var unit = ((Time64Type)Data.DataType).Unit;
+            return unit switch
+            {
+                TimeUnit.Second => value * 1_000,
+                TimeUnit.Millisecond => value,
+                TimeUnit.Microsecond => value / 1_000,
+                TimeUnit.Nanosecond => value / 1_000_000,
+                _ => throw new InvalidDataException($"Unsupported time unit for Time64Type: {unit}")
+            };
+        }
+
+        /// <summary>
+        /// Get the time at the specified index as microseconds
+        /// </summary>
+        /// <param name="index">Index at which to get the date.</param>
+        /// <returns>Returns a long, or <c>null</c> if there is no object at that index.
+        /// </returns>
+        public long? GetMicroSeconds(int index)
+        {
+            long? value = GetValue(index);
+            if (value == null)
+            {
+                return null;
+            }
+
+            var unit = ((Time64Type)Data.DataType).Unit;
+            return unit switch
+            {
+                TimeUnit.Second => value * 1_000_000,
+                TimeUnit.Millisecond => value * 1_000,
```

Comment:
```suggestion
```

According to 

https://github.com/apache/arrow/blob/25e0dd488ab60417f8f453f648e6ecfeb058f01e/format/Schema.fbs#L218-L220

Time64 only supports micro and nano seconds